qa/smoke-test-http/src/test/java/org/elasticsearch/http/ContextAndHeaderTransportIT.java

```diff
@@ -282,21 +287,20 @@ private Client transportClient() {
 
     public static class ActionLoggingPlugin extends Plugin implements ActionPlugin {
 
-        @Override
-        public Collection<Module> createGuiceModules() {
-            return Collections.<Module>singletonList(new ActionLoggingModule());
-        }
+        private final SetOnce<LoggingFilter> loggingFilter = new SetOnce<>();
 
         @Override
-        public List<Class<? extends ActionFilter>> getActionFilters() {
-            return singletonList(LoggingFilter.class);
+        public Collection<Object> createComponents(Client client, ClusterService clusterService, ThreadPool threadPool,
+                                                   ResourceWatcherService resourceWatcherService, ScriptService scriptService,
+                                                   NamedXContentRegistry xContentRegistry, Environment environment,
+                                                   NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry) {
+            loggingFilter.set(new LoggingFilter(clusterService.getSettings(), threadPool));
```

Comment:
I think this is a sign that `getActionFilters` maybe should take `ThreadPool` as an argument.

Reply:
As this is meant to be extended by plugins, it's unclear what parameters would be useful to them. I think it's best to keep the interface lean.